TLM/TLM/Manager/Impl/VehicleBehaviorManager.cs

```diff
@@ -347,11 +347,12 @@ public class VehicleBehaviorManager : AbstractCustomManager, IVehicleBehaviorMan
 
                                     // If you can turn preferred and you're not going straight, continue
                                     SegmentGeometry currentSegGeo = SegmentGeometry.Get(uCurrentSegment);
+                                    SegmentEndGeometry currentSegEndGeo = currentSegGeo.EndNodeGeometry;
```

Comment:
Are you sure it's EndNodeGeometry?

Reply:
```suggestion
                                  SegmentEndGeometry currentSegEndGeo = currentSegGeo.GetEnd(targetNodeId);
```

Reply:
For me, with this it still working as expected

Reply:
KMS!

Okay, I have to ask this. What is the difference between a Segment and a SegmentEnd? Also, what does this "startNode" option do/mean (if I do `currentSegGeo.GetEnd(startNode);`, I need the startNode property).

Reply:
Segment has obviously two ends (connected nodes) so GetEnd(node) will get you proper end (imagine that you are in that vehicle approaching node - now you can easily decide which is left and right outgoing segment)